pkg/controller/operands/cdi_test.go

```diff
@@ -402,6 +413,183 @@ var _ = Describe("CDI Operand", func() {
 				Message: "CDI is degraded: Bar",
 			}))
 		})
+
+		Context("Unsupported Annotation", func() {
+			It("Should create CDI object with changes from the annotation", func() {
+				hco.Annotations = map[string]string{common.JSONPatchCDIAnnotationName: `[
+					{
+						"op": "add",
+						"path": "/config/featureGates/-",
+						"value": "fg1"
+					},
+					{
+						"op": "add",
+						"path": "/config/filesystemOverhead",
+						"value": {"global": "50", "storageClass": {"AAA": "75", "BBB": "25"}}
+					}
+				]`}
+
+				cdi, err := NewCDI(hco)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cdi.Spec.Config.FeatureGates).To(HaveLen(2))
+				Expect(cdi.Spec.Config.FeatureGates).To(ContainElement("fg1"))
+				Expect(cdi.Spec.Config.FilesystemOverhead).ToNot(BeNil())
+				Expect(cdi.Spec.Config.FilesystemOverhead.Global).Should(BeEquivalentTo("50"))
+				Expect(cdi.Spec.Config.FilesystemOverhead.StorageClass).To(HaveLen(2))
+				Expect(cdi.Spec.Config.FilesystemOverhead.StorageClass["AAA"]).Should(BeEquivalentTo("75"))
+				Expect(cdi.Spec.Config.FilesystemOverhead.StorageClass["BBB"]).Should(BeEquivalentTo("25"))
+			})
+
+			It("Should fail to create CDI object with wrong jsonPatch", func() {
+				hco.Annotations = map[string]string{common.JSONPatchCDIAnnotationName: `[
+					{
+						"op": "notExists",
+						"path": "/config/featureGates/-",
+						"value": "fg1"
+					}
+				]`}
+
+				_, err := NewCDI(hco)
+				Expect(err).To(HaveOccurred())
+			})
+
+			It("Ensure func should create CDI object with changes from the annotation", func() {
```

Comment:
What's the significant distinction between the "Ensure func...) test cases and these the directly invoke `NewCDI()`?
Is it worth the almost-duplicated test cases?

Reply:
These are unit test. One checks the `NewCDI` func and one checks the ensure func. Not sure I get the comment. We need to make sure that `NewCDI` works and that the `ensure` uses it corrctlly.